### PR TITLE
Add local resource paths to applications specs.

### PR DESCRIPTION
**Pull request by ExternalReality**
Charmers can now specify a path for the value of the resource property.

### CODE REVIEW

File: bundledata_test.go [hunk 1]
```diff
@@ -110,7 +110,7 @@ var parseTests = []struct {
 					"db":      "db",
 					"website": "public",
 				},
-				Resources: map[string]int{
+				Resources: map[string]interface{}{
```
Comment (wallyworld) [inline bundledata_test.go]: Let's also add a string value

File: bundledata_test.go [hunk 2]
```diff
@@ -241,6 +241,15 @@ func (*bundleDataSuite) TestCodecRoundTrip(c *gc.C) {
 			var bd charm.BundleData
 			err = codec.Unmarshal(data, &bd)
 			c.Assert(err, gc.IsNil)
+
+			for _, app := range bd.Applications {
+				for resName, res := range app.Resources {
+					if val, ok := res.(float64); ok {
+						app.Resources[resName] = int(val)
```
Comment (wallyworld) [inline bundledata_test.go]: As per the above comment, the test needs to include a string value

**Comment by wallyworld** [comment]
$$merge$$

**Comment by ExternalReality** [comment]
$$merge$$

**Comment by cmars** [comment]
$$merge$$

**Pull request merged**